lib/Rex/Commands/Rsync.pm

```diff
@@ -161,7 +161,7 @@ sub sync {
     Rex::Logger::debug("Downloading $source -> $dest");
     push @rsync_cmd, "rsync -rl --verbose --stats $params ";
 
-    $source = $auth->{user} . "\@$servername:$source";
+    $source = $auth->{user} . "\@$servername:$source" if $auth;
```

Comment:
I wonder if the issue could be solved by moving this line into the `if (!local_connection)` branch, instead of adding the `$auth` checks.

Authentication only seems to be important for non-local conections, and we also only need to prefix `$source` with the `user@server:` string for non-local connections.